wp-job-manager-functions.php

```diff
@@ -1712,3 +1712,21 @@ function job_manager_get_salary_unit_options( $include_empty = true ) {
 	 */
 	return apply_filters( 'job_manager_get_salary_unit_options', $options, $include_empty );
 }
+
+/**
+ * Delete filled_jobs_transient and expired_jobs_transient transients when a job listing is saved or updated.
+ *
+ * @param int     $post_id  The ID of the post being saved.
+ * @param WP_Post $post The post object being saved.
+ */
+function delete_filled_job_listing_transient_on_post_meta_update( int $post_id, WP_Post $post ): void {
+
+	if ( 'job_listing' === $post->post_type ) {
+
+		if ( '1' === get_post_meta( $post_id, '_filled', true ) ) {
+			delete_transient( 'hide_filled_jobs_transient' );
+		}
+	}
+
+}
+add_action( 'save_post', 'delete_filled_job_listing_transient_on_post_meta_update', 10, 2 );
```

Comment:
I'm curious about why this is separated from the other logic in this PR? Do we think this function is useful in the public API?

Reply:
@jom Ah, I hadn't realised the functions file was dedicated to the API. I'll move this to post-types.php.

One thing I'm having difficulty with is getting the method to trigger on `updated_post_meta`. What I'd want to do ideally, following Peter's comment above, is:

1. Set the transient
2. If a job has the `_filled` meta updated, either add that ID to the transient or remove it

Clearing it on every post save is far too much and makes the transient pointless imo.

Reply:
@jom So I've learned that I was doing a check like:

```
if ( '_filled' !== $meta_key || 'job_listing' !== get_post_type( $object_id ) ) {
	return;
}
```

But in debugging, I noticed that `$meta_key` is only ever returning `_edit_lock`, so what actually works is:

```
if ( '_edit_lock' !== $meta_key || 'job_listing' !== get_post_type( $object_id ) ) {
	return;
}
```

But I don't imagine checking against `_edit_lock` is all that useful. I'm stumped on how to check that the `_filled` meta was changed. I've committed what I have now at https://github.com/Automattic/WP-Job-Manager/pull/2449/commits/de746c49aacb3a010e644be3b12e6329b68401ac